src/components/Form/RadiobuttonGroup.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { Component } from 'react';
+import classnames from 'classnames';
+import { HtmlDiv, HtmlSpan } from '../../reset';
+import { VisuallyHidden } from '../Visually-hidden/Visually-hidden';
+import { idGenerator } from '../../utils/uuid';
+
+import { RadiobuttonProps } from './Radiobutton';
+
+const baseClassName = 'fi-radiobuttongroup';
+const radiobuttonGroupClassNames = {
+  container: `${baseClassName}_container`,
+  label: `${baseClassName}_label`,
+  hintText: `${baseClassName}_hintText`,
+};
+
+type Label = 'hidden' | 'visible';
+
+export interface RadiobuttonGroupProps {
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** Radiobutton or RadiobuttonDivider */
+  children: Array<React.ReactElement<RadiobuttonProps>>;
+  /**
+   * Hint text to be displayed under the label.
+   */
+  hintText?: string;
+  /** Label for group */
+  label: string;
+  /** Hide or show label. Label element is always present, but can be visually hidden.
+   * @default visible
+   */
+  labelMode?: Label;
+  /**
+   * Unique id
+   * If no id is specified, one will be generated using uuid
+   * @default uuidV4
+   */
+  id?: string;
+  /** Name for the group; this will be set to Radiobuttons */
+  name: string;
+  /** Value that is selected from group, should match with the value given to Radiobutton */
+  value?: string;
+  /** Value that is selected by default */
+  defaultValue?: string;
+}
+
+export interface RadiobuttonGroupProviderState {
+  onRadiobuttonChange: (value: string) => void;
+}
+
+const defaultProviderValue: RadiobuttonGroupProviderState = {
+  onRadiobuttonChange: () => null,
+};
+
+const { Provider, Consumer: RadiobuttonGroupConsumer } = React.createContext(
+  defaultProviderValue,
+);
+
+const RadiobuttonGroupItems = (
+  children: Array<React.ReactElement<RadiobuttonProps>>,
+  groupName: string,
+  selectedValue?: string,
+) =>
+  React.Children.map(
+    children,
+    (child: React.ReactElement<RadiobuttonProps>) => {
+      if (React.isValidElement(child) && child.props.value) {
```

Comment:
`&& child.props.value` condition is here to detect that the child is a Radiobutton and **not** a RadiobuttonDivider.
If there is more elegant and reliable way to make it better, we should change this.

Reply:
https://stackoverflow.com/a/53138782 could this work for detecting child types?

Reply:
That way didn't seem to be possible. But the following does work, but seems little wonky as required to import from core side. (_more circulars_)

```jsx
import { Radiobutton as CoreRadiobutton } from '../../core/Form/Radiobutton/Radiobutton';
...
if (React.isValidElement(child) && child.type === CoreRadiobutton) {
```

Should we change it like that ^^ ?

Reply:
This might become an issue later on, but I guess it's working for now. I'll leave it up to you to decide if we should leave it as is or pursue some other alternative.

Reply:
Currently there is no reliable way for checking the typing of child elements. Text and components etc. can be checked, but that does not provide additional value for now.

Reply:
Left a comment to code to clarify this.